packages/cli/src/controller/codegen-controller.ts

```diff
@@ -28,55 +41,128 @@ export interface ProcessedField {
   required: boolean;
   indexed: boolean;
   unique?: boolean;
+  isArray: boolean;
+  isJsonInterface: boolean;
+}
+
+export async function generateJsonInterfaces(projectPath, schema: string) {
+  const typesDir = path.join(projectPath, TYPE_ROOT_DIR);
+  const jsonObjects = getAllJsonObjects(schema);
+  const jsonInterfaces = jsonObjects.map((r) => {
+    const object = setJsonObjectType(r, jsonObjects);
+    const fields = processFields('jsonField', object.name, object.fields);
+    return {
+      interfaceName: object.name,
+      fields,
+    };
+  });
+
+  if (jsonInterfaces.length !== 0) {
+    const interfaceTemplate = {
+      props: {
+        jsonInterfaces,
+      },
+      helper: {
+        upperFirst,
+      },
+    };
+    try {
+      await renderTemplate(INTERFACE_TEMPLATE_PATH, path.join(typesDir, `interfaces.ts`), interfaceTemplate);
+      exportTypes.interfaces = true;
+    } catch (e) {
+      throw new Error(`When render json interfaces having problems.`);
+    }
+  }
 }
 
 export function processFields(
+  type: string,
   className: string,
-  fields: GraphQLEntityField[],
-  indexFields: GraphQLEntityIndex[]
+  fields: (GraphQLEntityField | GraphQLJsonFieldType)[],
+  indexFields?: GraphQLEntityIndex[]
 ): ProcessedField[] {
   const fieldList: ProcessedField[] = [];
   for (const field of fields) {
-    const newType = transformTypes(className, field.type.toString());
-    if (!newType) {
-      throw new Error(`Undefined type ${field.type.toString()} in Schema ${className}`);
-    }
-    const [indexed, unique] = indexFields.reduce<[boolean, boolean]>(
-      (acc, indexField) => {
-        if (indexField.fields.includes(field.name)) {
-          acc[0] = true;
-          if (indexField.fields.length === 1 && indexField.unique) {
-            acc[1] = true;
-          } else if (indexField.unique === undefined) {
-            acc[1] = false;
-          }
-        }
-        return acc;
-      },
-      [false, undefined]
-    );
-    fieldList.push({
+    const injectField = {
       name: field.name,
-      type: newType,
       required: !field.nullable,
-      indexed,
-      unique,
-    });
+      isArray: field.isArray,
+    } as ProcessedField;
+    if (type === 'entity' && indexFields) {
+      const [indexed, unique] = indexFields.reduce<[boolean, boolean]>(
```

Comment:
these several lines need to run for type: entity no matter indexFields exists or not.
though you can give indexFields a default value `[]`